nixos/modules/services/networking/iperf.nix

```diff
@@ -0,0 +1,92 @@
+{ config, lib, pkgs, ... }: with lib;
+let
+  cfg = config.services.iperf3;
+
+  api = {
+    enable = mkEnableOption "iperf3 network throughput testing server";
+    port = mkOption {
+      type        = types.int;
```

Comment:
You could use the newish `types.ints.u16` to prevent people trying to assign a port outside of the valid range (have seen people do this)